CDCgov/prime-reportstream
# 7558 UP store filter expression errors

This PR adds logging support for errors found while evaluating receiver filter expressions in the Universal Pipeline, without making them viewable on the Submissions API.

Test Steps:
1. On your `organizations.yml` file, change a `qualityFilter:` entry for a receiver to an unparsable expression (e.g. `#2!dej2+1))k))(`) . Make sure to run `reloadSettings` on Gradle after saving the changes.
2. Submit a Full ELR report via Postman (or whichever you use)
3. After receiving a `submissionId`, use that to check the History API until the response includes a list of `destinations`.
4. On receiving a response, scroll to the bottom of the output. Check the `errors` array and verify it's empty.
5. Go to the database and check in the `action_log` table for a new entry that matches the issue with the given expression.
6. Repeat steps 1-5, but this time use an expression that returns a non-Boolean value (e.g. `1+2`). Verify the new error text looks correct.


## Changes
- Add filter logging for any errors during the `evaluateCondition` step on the Universal Pipeline
- Use simplified messages on the API-viewable data

## Linked Issues
- #7558
